app/full_text_search.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python2.7
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+
+from google.appengine.api import search as appengine_search
+
+import model
+
+# The index name for full text search
+PERSON_FULL_TEXT_INDEX_NAME = 'person_information'
+
+def search(repo, query_txt, max_results):
```

Comment:
As chatted offline, make sure this method only returns results in the given repository. Also add test for it.